coordinator/statement_executor.go

```diff
@@ -932,6 +934,98 @@ func (e *StatementExecutor) executeShowSubscriptionsStatement(stmt *influxql.Sho
 	return rows, nil
 }
 
+func (e *StatementExecutor) executeShowTagKeys(q *influxql.ShowTagKeysStatement, ctx *query.ExecutionContext) error {
+	var dis []meta.DatabaseInfo
+	if q.Database != "" {
+		di := e.MetaClient.Database(q.Database)
+		if di == nil {
+			return fmt.Errorf("database not found: %s", q.Database)
+		} else if di.DefaultRetentionPolicy == "" {
+			return fmt.Errorf("database %s does not have default retention policy", q.Database)
+		}
+		dis = []meta.DatabaseInfo{*di}
+	} else {
+		dis = e.MetaClient.Databases()
```

Comment:
I don't think `SHOW TAG VALUES` should work across databases.

Reply:
Fixed in 156f25a.